src/frontend/app/store/helpers/autoscaler-helpers.ts

```diff
@@ -0,0 +1,94 @@
+
+export function transformArrayToMap(newPolicy, timezone) {
+  if (timezone === undefined || timezone === '') {
+    timezone = moment.tz.guess();
+  }
+  newPolicy.scaling_rules_map = {};
+  newPolicy.scaling_rules_form = [];
+  if (newPolicy.scaling_rules) {
+    newPolicy.scaling_rules.map((trigger) => {
+      initIfUndefined(trigger.breach_duration_secs, PolicyDefaultSetting.breach_duration_secs_default);
+      initIfUndefined(trigger.cool_down_secs, PolicyDefaultSetting.cool_down_secs_default);
+      pushAndSortTrigger(newPolicy.scaling_rules_map, trigger.metric_type, trigger);
+    });
+  }
+  Object.keys(newPolicy.scaling_rules_map).map((metricName) => {
+    ScaleTypes.map((triggerType) => {
+      if (newPolicy.scaling_rules_map[metricName][triggerType]) {
+        newPolicy.scaling_rules_map[metricName][triggerType].map((trigger) => {
+          newPolicy.scaling_rules_form.push(trigger);
+        });
+      }
+    });
+  });
+  initIfUndefined(newPolicy.schedules, {timezone});
+  initIfUndefined(newPolicy.schedules.recurring_schedule, []);
+  initIfUndefined(newPolicy.schedules.specific_date, []);
+  return newPolicy;
+}
+
+export function transformMapToArray(newPolicy) {
+  if (newPolicy.scaling_rules_form) {
+    const scaling_rules = [];
+    newPolicy.scaling_rules_form.map((trigger) => {
+      deleteIf(trigger, 'breach_duration_secs', trigger.breach_duration_secs === PolicyDefaultSetting.breach_duration_secs_default);
+      deleteIf(trigger, 'cool_down_secs', trigger.cool_down_secs === PolicyDefaultSetting.cool_down_secs_default);
+      scaling_rules.push(trigger);
+    });
+    if (scaling_rules.length > 0) {
+      newPolicy.scaling_rules = scaling_rules;
+    }
+  }
+  delete newPolicy['scaling_rules_form'];
+  delete newPolicy['scaling_rules_map'];
+  if (newPolicy.schedules) {
+    deleteIf(
+      newPolicy.schedules,
+      'recurring_schedule',
+      newPolicy.schedules.recurring_schedule && newPolicy.schedules.recurring_schedule.length === 0);
+    deleteIf(
+      newPolicy.schedules,
+      'specific_date',
+      newPolicy.schedules.specific_date && newPolicy.schedules.specific_date.length === 0);
+    deleteIf(newPolicy, 'schedules', !newPolicy.schedules.recurring_schedule && !newPolicy.schedules.specific_date);
+  }
+  return newPolicy;
+}
+
+function pushAndSortTrigger(map, metricName, newTrigger) {
+  const scaleType = LowerOperators.indexOf(newTrigger.operator) >= 0 ? ScaleTypes[0] : ScaleTypes[1];
+  if (!map[metricName]) {
+    map[metricName] = {};
+  }
+  if (!map[metricName][scaleType]) {
+    map[metricName][scaleType] = [];
+  }
+  for (let i = 0; i < map[metricName][scaleType].length; i++) {
+    if (newTrigger.threshold > map[metricName][scaleType][i].threshold) {
+      map[metricName][scaleType].splice(i, 0, newTrigger);
+      return;
+    }
+  }
+  map[metricName][scaleType].push(newTrigger);
+}
+
+function initIfUndefined(rawData, defaultData) {
+  if (!rawData || rawData === '') {
```

Comment:
I'm a bit confused by this, I don't think rawData will change outside of the function. Additionally (iif `rawData === 0` counts as undefined) this function can be removed and the `a || b` syntax could be used. 
So `rawData || defaultData` would be the same as
```
if (!!rawData) {
    return defaultData
}
```